BlackNectar/BlackNectar/StoresDataStruct.swift

```diff
@@ -11,63 +11,65 @@ import CoreLocation
 import Foundation
 import UIKit
 
-// Creating data structure for JSON Request
-struct StoresInfo {
+// Creates data structure for JSON Request
+struct Stores {
     
     let storeName: String
     let location: CLLocationCoordinate2D
-    let address: NSDictionary
+    let address: Address
     let storeImage: URL
     let isFarmersMarket: Bool
     
     var notFarmersMarket: Bool { return !isFarmersMarket }
     
-    static func fromJson(dictionary: NSDictionary) -> StoresInfo? {
-        
-        guard let storeName = dictionary ["store_name"] as? String,
-            let location = dictionary ["location"] as? NSDictionary,
-            let address = dictionary ["address"] as? NSDictionary,
-            let storeType = dictionary ["main_image_url"] as? String,
+    static func getStoreJsonData(from storeDictionary: NSDictionary) -> Stores? {
+    
+        guard let storeName = storeDictionary ["store_name"] as? String,
+            let addressJSON = storeDictionary ["address"] as? NSDictionary,
+            let storeType = storeDictionary ["main_image_url"] as? String,
             let storeImage = URL(string: storeType)
 
             else {
                 
-                LOG.error("Guard Failed on fromJson method")
+                LOG.error("Failed on getting Store JSON Data")
                 return nil
                 
         }
         
-        guard let coordinatesDictionary = dictionary ["location"] as? NSDictionary else { return nil }
+        guard let addressObject = Address(from: addressJSON) else { return nil }
+        
+        guard let coordinatesDictionary = storeDictionary ["location"] as? NSDictionary else { return nil }
         guard let latitude = coordinatesDictionary ["latitude"] as? CLLocationDegrees else { return nil }
         guard let longitude = coordinatesDictionary ["longitude"] as? CLLocationDegrees else { return nil }
         let coordinatesObject = CLLocationCoordinate2D(latitude: latitude, longitude: longitude)
         
-        let isFarmersMarket: Bool = dictionary["is_farmers_market"] as? Bool ?? false
+        let isFarmersMarket: Bool = storeDictionary["is_farmers_market"] as? Bool ?? false
         
-        return StoresInfo(storeName: storeName, location: coordinatesObject, address: address, storeImage: storeImage, isFarmersMarket: isFarmersMarket)
+        return Stores(storeName: storeName, location: coordinatesObject, address: addressObject, storeImage: storeImage, isFarmersMarket: isFarmersMarket)
         
     }
     
 }
 
-struct AddressInfo {
+struct Address {
     
-    let street: String
-    let city: String
-    let state: String
+    let addressLineOne: String?
```

Comment:
Having issues with this one 
At the moment the only way I can make it work is if I make all of them optionals
could use some help